packages/plugin/package.json

```diff
@@ -39,9 +39,15 @@
     "typecheck": "tsc --noEmit"
   },
   "peerDependencies": {
+    "@apollo/federation-internals": "^2",
     "eslint": ">=8.44.0",
     "graphql": "^16"
   },
+  "peerDependenciesMeta": {
+    "@apollo/federation-internals": {
+      "optional": true
+    }
+  },
```

Comment:
I think people should explicitly install this dependency if they want to validate federation schema, there is no need to add to dependencies